### PR TITLE
New home screen

### PR DESCRIPTION
**Pull request by MatheusUehara**
Created new home screen with navigation. 

Created a static splash screen for the app. 

Added bottomBarNavigation on App.

Some adjusts in repository method parameters order

Decreased offset in requests from 50 to 20 heroes.

### CODE REVIEW

**Comment by daividvleal** [comment]
resolve conflicts first

File: app/src/main/kotlin/com/br/myfavoritehero/features/listCharacter/ListHeroesFragment.kt [hunk 1]
```diff
@@ -51,13 +64,22 @@ class ListHeroesActivity : AppCompatActivity(), HeroEventListener {
                     shimmer_view_container.stopShimmerAnimation()
 
                     listHeroes.setHasFixedSize(true)
-                    val layoutManager = LinearLayoutManager(this)
+                    val layoutManager = LinearLayoutManager(activity)
                     listHeroes.layoutManager = layoutManager
                     heroAdapter = HeroAdapter(stateModel.model!!, this)
```
Comment (daividvleal) [inline app/src/main/kotlin/com/br/myfavoritehero/features/listCharacter/ListHeroesFragment.kt]: don't use !!, high risk

File: app/src/main/res/layout/activity_bottom_navigation.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        app:defaultNavHost="true"
+        app:layout_constraintBottom_toTopOf="@+id/navigation"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:navGraph="@navigation/nav_graph" />
+
+    <com.google.android.material.bottomnavigation.BottomNavigationView
+        android:id="@+id/navigation"
+        android:layout_width="0dp"
```
Comment (daividvleal) [inline app/src/main/res/layout/activity_bottom_navigation.xml]: use dimen file

File: app/src/main/res/layout/hero_list_loading.xml [hunk 3]
```diff
@@ -3,14 +3,14 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
     android:id="@+id/shimmer_view_container"
     android:layout_width="match_parent"
-    android:layout_height="match_parent"
+    android:layout_height="wrap_content"
     android:layout_gravity="center"
     android:orientation="vertical"
     app:duration="800">
```
Comment (daividvleal) [inline app/src/main/res/layout/hero_list_loading.xml]: use integers file, create new file in values like strings

File: app/src/main/res/menu/navigation.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <item
+        android:id="@+id/listHeroesFragment"
+        android:icon="@drawable/heroes"
+        android:orderInCategory="100"
+        android:title="@string/heroes"
+        app:showAsAction="ifRoom" />
+
+    <!--<item-->
```
Comment (daividvleal) [inline app/src/main/res/menu/navigation.xml]: erase it

File: build.gradle [hunk 5]
```diff
@@ -30,6 +30,7 @@ buildscript {
         junit="4.12"
         runner="1.1.1"
         espresso_core="3.1.1"
+        navigation="1.0.0-alpha01"
```
Comment (daividvleal) [inline build.gradle]: are we gonna use navigation, shouldn't we discuss it first? the benefits and problems?

File: app/build.gradle [hunk 6]
```diff
@@ -60,6 +60,10 @@ dependencies {
     //Timber
     implementation "com.jakewharton.timber:timber:$timber"
 
+    //Navigation
+    implementation "android.arch.navigation:navigation-fragment-ktx:$navigation"
```
Comment (daividvleal) [inline app/build.gradle]: are we gonna use navigation, shouldn't we discuss it first? the benefits and problems?

File: app/src/androidTest/java/com/br/myfavoritehero/ExampleInstrumentedTest.kt [hunk 7]
```diff
@@ -0,0 +1,24 @@
+package com.br.myfavoritehero
```
Comment (daividvleal) [inline app/src/androidTest/java/com/br/myfavoritehero/ExampleInstrumentedTest.kt]: Is this file necessary?

File: app/src/main/kotlin/com/br/myfavoritehero/features/heroDetails/DetailHeroActivity.kt [hunk 8]
```diff
@@ -32,45 +32,45 @@ class DetailHeroActivity : AppCompatActivity() {
...
-                favorited = !favorited
-            }else{
+                !favorited
+            } else {
                 fab_favorite_hero.setImageResource(R.drawable.un_favorite)
-                SharedPreferencesHelper.setFavorite(this,hero.id)
+                SharedPreferencesHelper.setFavorite(this, hero.id)
                 Snackbar.make(scroll_view, R.string.favorited, Snackbar.LENGTH_SHORT).show()
-                favorited = !favorited
+                !favorited
```
Comment (daividvleal) [inline app/src/main/kotlin/com/br/myfavoritehero/features/heroDetails/DetailHeroActivity.kt:53]: Great job!

File: app/src/main/kotlin/com/br/myfavoritehero/util/ThumbnailExtension.kt [hunk 9]
```diff
@@ -5,14 +5,14 @@ import com.br.myfavoritehero.util.Constants.portrait_fantastic
 import com.br.myfavoritehero.util.Constants.separator
 import com.br.myfavoritehero.util.Constants.type
 
-fun String.getLargePortraitThumbnail(): String{
-    return "$this$separator$portrait_fantastic$type"
+fun String.getLargePortraitThumbnail(): String {
+    return "${this.replace("http://", "https://")}$separator$portrait_fantastic$type"
```
Comment (daividvleal) [inline app/src/main/kotlin/com/br/myfavoritehero/util/ThumbnailExtension.kt]: Should it be constants?

File: app/src/main/kotlin/com/br/myfavoritehero/util/ThumbnailExtension.kt [hunk 10]
```diff
@@ -5,14 +5,14 @@ import com.br.myfavoritehero.util.Constants.portrait_fantastic
...
 }
 
-fun String.getLargeLandscapeThumbnail(): String{
-    return "$this$separator$landscape_amazing$type"
+fun String.getLargeLandscapeThumbnail(): String {
+    return "${this.replace("http://", "https://")}$separator$landscape_amazing$type"
```
Comment (daividvleal) [inline app/src/main/kotlin/com/br/myfavoritehero/util/ThumbnailExtension.kt]: Should it be constants?

File: app/src/main/kotlin/com/br/myfavoritehero/util/ThumbnailExtension.kt [hunk 11]
```diff
@@ -5,14 +5,14 @@ import com.br.myfavoritehero.util.Constants.portrait_fantastic
...
 }
 
-fun String.getLargeThumbnail(): String{
-    return "$this$separator$landscape_amazing$type"
+fun String.getLargeThumbnail(): String {
+    return "${this.replace("http://", "https://")}$separator$landscape_amazing$type"
```
Comment (daividvleal) [inline app/src/main/kotlin/com/br/myfavoritehero/util/ThumbnailExtension.kt]: Should it be constants?

File: app/src/main/res/drawable/background_border.xml [hunk 12]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android" >
+
+    <stroke android:width="1dp" android:color="@color/white"/>
+
+    <corners
+        android:bottomLeftRadius="16dp"
```
Comment (daividvleal) [inline app/src/main/res/drawable/background_border.xml:7]: Dimens?

File: app/src/main/res/drawable/favorite.xml [hunk 13]
```diff
@@ -4,5 +4,5 @@
         android:width="24dp"
         android:viewportWidth="24"
         android:viewportHeight="24">
-    <path android:fillColor="#e53935" android:pathData="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z" />
+    <path android:fillColor="#FFFFFF" android:pathData="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z" />
```
Comment (daividvleal) [inline app/src/main/res/drawable/favorite.xml:7]: Colors?

File: app/src/main/res/drawable/heroes.xml [hunk 14]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:height="24dp"
+    android:width="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
```
Comment (daividvleal) [inline app/src/main/res/drawable/heroes.xml:6]: Dimens?

File: app/src/main/res/drawable/heroes.xml [hunk 15]
```diff
@@ -0,0 +1,8 @@
...
+    <path android:fillColor="#FFFFFF" android:pathData="M19,1.27C18.04,0.72 16.82,1.04 16.27,2C15.71,2.95 16.04,4.18 17,4.73C17.95,5.28 19.17,4.96 19.73,4C20.28,3.04 19.95,1.82 19,1.27M21.27,9.34L18.7,13.79L16.96,12.79L18.69,9.79L17.14,8.5L14,13.92V22H12V13.39L2.47,7.89L3.47,6.16L11.27,10.66L13.67,6.5L7.28,4.17L8,2.29L14.73,4.74L15,4.84C15.39,5 15.76,5.15 16.12,5.35L16.96,5.84C17.31,6.04 17.65,6.28 17.96,6.54L18.19,6.74L21.27,9.34Z" />
```
Comment (daividvleal) [inline app/src/main/res/drawable/heroes.xml:7]: Colors?

File: app/src/main/res/drawable/un_favorite.xml [hunk 16]
```diff
@@ -4,5 +4,5 @@
         android:width="24dp"
         android:viewportWidth="24"
         android:viewportHeight="24">
-    <path android:fillColor="#e53935" android:pathData="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z" />
+    <path android:fillColor="#FFFFFF" android:pathData="M12,17.27L18.18,21L16.54,13.97L22,9.24L14.81,8.62L12,2L9.19,8.62L2,9.24L7.45,13.97L5.82,21L12,17.27Z" />
```
Comment (daividvleal) [inline app/src/main/res/drawable/un_favorite.xml]: Colors?

File: app/src/main/res/layout/hero_item.xml [hunk 17]
```diff
@@ -1,58 +1,58 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_marginLeft="@dimen/margin_default"
+    android:layout_marginTop="@dimen/margin_default"
+    android:layout_marginRight="@dimen/margin_default"
+    android:layout_marginBottom="@dimen/margin_default"
+    android:orientation="vertical"
+    tools:cardElevation="2dp"
```
Comment (daividvleal) [inline app/src/main/res/layout/hero_item.xml]: Dimens?

File: app/src/main/res/layout/hero_item.xml [hunk 18]
```diff
@@ -1,58 +1,58 @@
...
-                app:layout_constraintBottom_toBottomOf="parent"
-                app:layout_constraintEnd_toStartOf="@+id/favoriteIcon"
-        />
+            android:id="@+id/heroName"
+            android:layout_width="@dimen/dp_0"
+            android:layout_height="wrap_content"
+            android:layout_marginEnd="@dimen/margin_default"
+            android:layout_marginRight="@dimen/margin_default"
+            android:ellipsize="end"
+            android:maxLines="1"
```
Comment (daividvleal) [inline app/src/main/res/layout/hero_item.xml]: Should be a grate idea create an integer file, to set this and other integer elements?

File: app/src/main/res/layout/hero_item_placeholder.xml [hunk 19]
```diff
@@ -1,58 +1,60 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_marginLeft="@dimen/margin_default"
+    android:layout_marginTop="@dimen/margin_default"
+    android:layout_marginRight="@dimen/margin_default"
+    android:layout_marginBottom="@dimen/margin_default"
+    android:orientation="vertical"
+    tools:cardElevation="2dp"
```
Comment (daividvleal) [inline app/src/main/res/layout/hero_item_placeholder.xml]: Dimens?

File: app/src/main/res/layout/hero_item_placeholder.xml [hunk 20]
```diff
@@ -1,58 +1,60 @@
...
-                    android:layout_marginRight="@dimen/margin_default"
-                    android:background="@color/shimmer_background"
-                    android:layout_width="@dimen/dp_0"
-                    android:layout_height="@dimen/dp_20"/>
+                android:layout_width="@dimen/dp_0"
+                android:layout_height="@dimen/dp_20"
+                android:layout_gravity="center_vertical"
+                android:layout_marginLeft="@dimen/margin_default"
+                android:layout_marginRight="@dimen/margin_default"
+                android:layout_weight="1"
```
Comment (daividvleal) [inline app/src/main/res/layout/hero_item_placeholder.xml]: Should be a grate idea create an integer file, to set this and other integer elements?

**Review by daividvleal — COMMENTED** [review]
The comments are not something that you have to follow everything, discussions are encouraged!

Also, the splash screen keeps open when we press the android back button!

File: app/src/main/res/layout/hero_item.xml (hunk 18 above)
Comment (MatheusUehara) [reply app/src/main/res/layout/hero_item.xml]: i create integers file and forget maxLines and remains magic numbers.... in next commits i will fix it.

File: app/src/main/res/layout/hero_item_placeholder.xml (hunk 20 above)
Comment (MatheusUehara) [reply app/src/main/res/layout/hero_item_placeholder.xml]: done

File: app/src/main/res/layout/hero_item_placeholder.xml (hunk 19 above)
Comment (MatheusUehara) [reply app/src/main/res/layout/hero_item_placeholder.xml]: done

File: app/src/main/res/layout/hero_item.xml (hunk 18 above)
Comment (MatheusUehara) [reply app/src/main/res/layout/hero_item.xml]: done

File: app/src/main/res/layout/hero_item.xml (hunk 17 above)
Comment (MatheusUehara) [reply app/src/main/res/layout/hero_item.xml]: done

File: app/src/main/res/drawable/un_favorite.xml (hunk 16 above)
Comment (MatheusUehara) [reply app/src/main/res/drawable/un_favorite.xml]: done

File: app/src/main/kotlin/com/br/myfavoritehero/util/ThumbnailExtension.kt (hunk 11 above)
Comment (MatheusUehara) [reply app/src/main/kotlin/com/br/myfavoritehero/util/ThumbnailExtension.kt]: done

File: app/src/main/kotlin/com/br/myfavoritehero/util/ThumbnailExtension.kt (hunk 10 above)
Comment (MatheusUehara) [reply app/src/main/kotlin/com/br/myfavoritehero/util/ThumbnailExtension.kt]: done

File: app/src/main/kotlin/com/br/myfavoritehero/util/ThumbnailExtension.kt (hunk 9 above)
Comment (MatheusUehara) [reply app/src/main/kotlin/com/br/myfavoritehero/util/ThumbnailExtension.kt]: done

File: app/src/main/kotlin/com/br/myfavoritehero/features/heroDetails/DetailHeroActivity.kt (hunk 8 above)
Comment (MatheusUehara) [reply app/src/main/kotlin/com/br/myfavoritehero/features/heroDetails/DetailHeroActivity.kt:53]: :D

File: app/src/androidTest/java/com/br/myfavoritehero/ExampleInstrumentedTest.kt (hunk 7 above)
Comment (MatheusUehara) [reply app/src/androidTest/java/com/br/myfavoritehero/ExampleInstrumentedTest.kt]: nop, i believe this is generated class...

**Comment by MatheusUehara** [comment]
> The comments are not something that you have to follow everything, discussions are encouraged!
> 
> Also, the splash screen keeps open when we press the android back button!

Last commit fix back button going to splash screen :D
